titus-api/src/main/java/com/netflix/titus/api/agent/model/AgentFunctions.java

```diff
@@ -47,4 +48,18 @@ private AgentFunctions() {
             return Optional.empty();
         }
     }
+
+    public static Function<AgentInstanceGroup, AgentInstanceGroup> withId(String newId) {
+        return ig -> ig.toBuilder().withId(newId).build();
+    }
+
+    public static Function<AgentInstanceGroup, AgentInstanceGroup> inState(InstanceGroupLifecycleState state) {
+        return ig -> ig.toBuilder().withLifecycleStatus(
+                InstanceGroupLifecycleStatus.newBuilder()
+                        .withState(state)
+                        .withTimestamp(System.currentTimeMillis())
```

Comment:
Shouldn't this be based on the clock abstraction?

Reply:
Fixed